### ISSUE TITLE
performance difference between this repo and origin paper of 'social lstm'

### ISSUE THREAD

**Issue by ZongweiZhou1** [vvanirudh/social-lstm-pytorch#11]
@vvanirudh  Thanks for this repo, it's a wondeful work. 

But i note the performances of social lstm reported in 'social attention' and 'social lstm' have a big difference. For example, the final displacement error of social lstm reported in 'social lstm' is 0.61 (Table 1) while it is 3.32 in 'social attention'. 

Is there any difference operation in the evaluation process? Or the performance of 'social lstm' is impossible to reproduce?

**Comment by avijit9** [comment]
Why have you closed this issue? Are you able to reproduce? If yes, can you highlight the part you changed?

**Comment by ZongweiZhou1** [comment]
@avijit9  Well, I dont find the code for evaluation, but I find the paper said that 

 > In our evaluation, we also included the prediction errors of pedestrians for whom we observed fewer than T_{obs} time-steps as they entered the crowd at a later time.... This is one of the primary reasons for the difference in our results of Social LSTM compared to that from the original paper.

Thus, I closed this issue to evaluate the performace as this paper said.

Do you have the evaluation codes?

**Comment by avijit9** [comment]
The evaluation code is provided, right? sample.py? 
Even if the code includes the pedestrians which we observe less than 8 timesteps, I am not able to reproduce the results reported in the paper. That is why I asked.

**Comment by ZongweiZhou1** [comment]
@avijit9  
Thanks a lot, sample.py is omitted by my mistake :(
I am training the model now and i'll come back when i get the results.

By the way, I found there were 'Dropout Layer's  defined in the net, but the net's mode hadnot set to be eval in validation. Does this have some effects on performance?

**Comment by ZongweiZhou1** [comment]
@avijit9 
I just test the sample.py on ZARA2 dataset, and I got the total mean error: 0.2965, the total final error: 0.3764. It is not exactly same with the origin paper 'Social LSTM', but they have the same order of magnitude. 
I dont understand is why the 'Final Displacement Error' reported in 'Social attention' are so larger??

ps. I change the net mode to eval in sample.py to close the Dropout layer.

**Comment by avijit9** [comment]
Here is the catch. The result you are getting are in pixel space. The result reported in the paper are in meter. 

Even if I remove the dropout layer, the situation remains the same.

**Comment by zhangpur** [comment]
May similar situation as you. Have you solved this？

**Comment by avijit9** [comment]
@vvanirudh  can you please help us with the solution? As the paper is already accepted to a conference we need to compare our algorithm with yours.
